### PR TITLE
#23847 Add configuration to Reach Page type Goal

### LINKED ISSUE #23847: Add to REACH_PAGE Goal the configurations
### Proposed Objective

Core Improvement (Features)

### Proposed Priority

Priority 3 - Average

### User Story

As a user, I want to be able to add the name of the REACH_PAGE Goal and set the next conditions:

- parameter
- - url (Mandatory)
- operator (Mandatory)
- - EQUALS or COINTAIN
- value (Mandatory)

### Acceptance Criteria

- Add a REACH_PAGE Goal and show a sidebar with the required inputs
- When saving it, go to the config page and show the changes.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs
Body 

```{
    "goals": {
        "primary": {
            "name": "asdasd",
            "type": "REACH_PAGE",
            "conditions": [
                {
                    "parameter": "url",
                    "operator": "EQUALS",
                    "value": "value"
                }
            ]
        }
    }
}

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Add the ability for the user to configure Reach Page (Define Name, y Conditions)

### Checklist
- [x] Tests
- [x] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
Reuse the sidebar 

### Screenshots
Original    
old-set-goal

New config
new-set-goal

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.spec.ts [hunk 1]
```diff
@@ -65,43 +75,102 @@ describe('DotExperimentsConfigurationGoalSelectComponent', () => {
...
+        spectator.component.form.setValue(formValues);
+        spectator.component.form.updateValueAndValidity();
+
+        spectator.detectComponentChanges();
+
+        expect(spectator.component.form.valid).toBeTrue();
+        expect(applyBtn.disabled).toBeFalse();
+    });
+
+    it('should call saveForm() when a item is selected and the button of apply is valid', () => {
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.spec.ts:137]: better, should call setSelectedGoal....

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts [hunk 2]
```diff
@@ -34,56 +44,128 @@ import { DotSidebarHeaderComponent } from '@shared/dot-sidebar-header/dot-sideba
...
+
     constructor(
         private readonly dotExperimentsConfigurationStore: DotExperimentsConfigurationStore
     ) {}
 
+    /**
+     * Get the conditions FormArray
+     *
+     * @return FormArray
+     */
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts:82]: missing, `memberof`

Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts:82]: Use https://marketplace.visualstudio.com/items?itemName=oouo-diogo-perdigao.docthis

File: core-web/libs/dot-primeng-theme-styles/src/scss/_variables.scss [hunk 3]
```diff
@@ -141,6 +139,7 @@ $font-size-small: 0.85rem;
 $font-size-x-small: 0.75rem;
 $font-weight-regular-bold: 400;
 $font-weight-semi-bold: 500;
+$font-weight-bold: 700;
```
Comment (fmontes) [inline core-web/libs/dot-primeng-theme-styles/src/scss/_variables.scss:142]: This is getting out of hand... I will check with Melissa to unify this.

**Review by fmontes — DISMISSED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiment-options/dot-experiment-options.component.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    toggleOption = (index: number) => {
+        if (this.expanded.has(index)) {
+            this.expanded.delete(index);
+        } else {
+            this.expanded.clear();
+            this.expanded.add(index);
+        }
+    };
+
+    /**
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiment-options/dot-experiment-options.component.ts:64]: fix documentation missing `memberof`

File: core-web/apps/dotcms-ui/src/app/test/dot-message-mock.pipe.ts [hunk 5]
```diff
@@ -5,7 +5,7 @@ import { Pipe, PipeTransform } from '@angular/core';
  * use only with tests
  */
 @Pipe({ name: 'dm', standalone: true })
-export class DotMessageMockPipe implements PipeTransform {
+export class DotMessagePipe implements PipeTransform {
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/test/dot-message-mock.pipe.ts:8]: The IDE when I try lo import show me this.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiment-options/directives/dot-experiment-option-content.directive.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * Get the content TemplateRef of the Option
+ */
+@Directive({
+    selector: '[dotOptionContent]'
+})
+export class DotExperimentOptionContentDirective {
+    constructor(public templateRef: TemplateRef<unknown>) {}
+}
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiment-options/directives/dot-experiment-option-content.directive.ts:11]: What is the goal of this `directive`? I know this directive is being used in an `ng-template`. However, I don't know where the `templateRef` is being used.

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling/dot-experiments-configuration-scheduling.component.spec.ts [hunk 7]
```diff
@@ -87,7 +87,7 @@ describe('DotExperimentsConfigurationSchedulingComponent', () => {
         expect(store.openSidebar).toHaveBeenCalledOnceWith(ExperimentSteps.SCHEDULING);
     });
 
-    it('should render the date range in the button when present', () => {
+    xit('should render the date range in the button when present', () => {
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling/dot-experiments-configuration-scheduling.component.spec.ts:90]: it's ok.. for the others, we need to mock `new Date()` here. Will do it on other PR.

**Review by hmoreras — APPROVED** [review]

**Pull request merged**